packages/xo-server-audit/src/index.js

```diff
@@ -245,6 +251,44 @@ class AuditXoPlugin {
       }))
   }
 
+  // See www-xo#344
+  async _automaticIntegrityCheck() {
+    const xo = this._xo
+    const chain = await xo.audit.getLastChain()
+
+    let lastHash, integrityCheckSuccess
+    if (chain !== null) {
+      const hashes = chain.hashes
+      lastHash = hashes[hashes.length - 1]
+
+      // check the integrity of all stored hashes
+      integrityCheckSuccess = await Promise.all(
+        hashes.map((oldest, key) =>
+          oldest !== lastHash
+            ? this._checkIntegrity({ oldest, newest: hashes[key + 1] })
+            : true
+        )
+      ).then(
+        () => true,
+        () => false
+      )
+    }
+
+    // generate a valid fingerprint of all stored records in case of a failure integrity check
+    const { oldest, newest, error } = await this._generateFingerprint({
+      oldest: integrityCheckSuccess ? lastHash : undefined,
+    })
+
+    if (chain === null || !integrityCheckSuccess || error !== undefined) {
+      await xo.audit.startNewChain({
+        oldest,
+        newest,
+      })
```

Comment:
Small formatting detail: make this object single line for homgeneity :slightly_smiling_face:

Reply:
By declaring a const?

Reply:
No, you should be able to make it single line, like the `extendLastChain` call.